routemaster/services/deliver.rb

```diff
@@ -36,14 +45,24 @@ def call
         error = nil
         start_at = Routemaster.now
         begin
-        # send data
+          _maybe_throttle!
+          @subscriber.attempting_delivery(start_at)
+
+          # send data
           response = _conn.post do |post|
             post.headers['Content-Type'] = 'application/json'
             post.body = Oj.dump(_data, mode: :compat)
           end
-          error = CantDeliver.new("HTTP #{response.status}") unless response.success?
+
+          unless response.success?
+            @subscriber.change_health_by(-2)
+            error = CantDeliver.new("HTTP #{response.status}")
+          end
+        rescue EarlyThrottle => e
+          error = _wrap_error(e)
```

Comment:
Wrapping `EarlyThrottle` in `CantDeliver` feel semantically a little wrong — the latter means we tried but failed; the former means we didn't even try. Not sure wrapping is required here; you could have an exception superclass `Deliver::Failure` (maybe with a common `:delay` accessor as per other comments), and rescue from taht in the job.